tests/test_empty_suffix.py

```diff
@@ -1,57 +1,60 @@
+import pytest
+
 import copier
 
 from .helpers import build_file_tree
 
 
-def test_empty_suffix(tmp_path_factory):
-    root = tmp_path_factory.mktemp("demo_empty_suffix")
+def test_empty_suffix(tmp_path_factory: pytest.TempPathFactory) -> None:
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
     build_file_tree(
         {
-            root
-            / "copier.yaml": """
+            (src / "copier.yaml"): (
+                """\
                 _templates_suffix: ""
                 name:
                     type: str
                     default: pingu
-            """,
-            root / "render_me": "Hello {{name}}!",
-            root / "{{name}}.txt": "Hello {{name}}!",
-            root / "{{name}}" / "render_me.txt": "Hello {{name}}!",
+                """
+            ),
+            (src / "render_me"): "Hello {{name}}!",
+            (src / "{{name}}.txt"): "Hello {{name}}!",
+            (src / "{{name}}" / "render_me.txt"): "Hello {{name}}!",
         }
     )
-    dest = tmp_path_factory.mktemp("dst")
-    copier.copy(str(root), dest, defaults=True, overwrite=True)
+    copier.copy(str(src), dst, defaults=True, overwrite=True)
 
-    assert not (dest / "copier.yaml").exists()
+    assert not (dst / "copier.yaml").exists()
 
-    assert (dest / "render_me").exists()
-    assert (dest / "pingu.txt").exists()
-    assert (dest / "pingu" / "render_me.txt").exists()
+    assert (dst / "render_me").exists()
+    assert (dst / "pingu.txt").exists()
+    assert (dst / "pingu" / "render_me.txt").exists()
 
     expected = "Hello pingu!"
-    assert (dest / "render_me").read_text() == expected
-    assert (dest / "pingu.txt").read_text() == expected
-    assert (dest / "pingu" / "render_me.txt").read_text() == expected
+    assert (dst / "render_me").read_text() == expected
+    assert (dst / "pingu.txt").read_text() == expected
+    assert (dst / "pingu" / "render_me.txt").read_text() == expected
```

Comment:
I repsect your position on this diff. But I really believe there is some value in harmonizing the tests a bit, also regarding the recurring creation code for the temporary template and subproject folders as well as the variable names for their paths. It removes variance in the tests where in fact none exists which makes it easier for new contributors to recognize recurring setup code. When several variations are used across the test suite, recognizing semantically identical code is more difficult.

If you really insist on keeping it as it is, I'll do it. But I genuinely think there is value in this harmonization.

Reply:
It's not an easy decision to take. On one hand, I obviously like prettier code. But OTOH, I already found 2 issues with the refactor, and the old motto "if something ain't broken, don't fix it" comes to my mind. I'm afraid that with the refactor we're introducing more non-obvious bugs, or at least skipping some other non-obvious corner cases that are being implicitly tested.

All my life I considered tests the only part of code that is "write and forget". I don't actually "maintain" tests. I just fix them when they're broken. And those breaks help me to know that the next release will be a major one, because something is broken. Also, writing tests is boring enough, so when a contributor contributes a test that is not diverging too much from the somehow modern way of testing copier, I just accept that. I mean: as long as the test is testing what it must test and goes green, it's mostly OK for me.

Putting all things in balance, sadly I think stability is weightier than beauty, so let me gladly ask you to please start reverting most of the diff. Let me do more comments on your specific questions below.

Reply:
Oh well ... Done. :shrug: